src/Microsoft.DotNet.ProjectModel/Files/PatternGroup.cs

```diff
@@ -12,6 +12,8 @@ namespace Microsoft.DotNet.ProjectModel.Files
 {
     public class PatternGroup
     {
+        private static readonly Dictionary<string, IEnumerable<string>> s_resolvedFilesCache = new Dictionary<string, IEnumerable<string>>();
```

Comment:
Without some deeper fixes to change the instantiation pattern of IncludeFilesResolver, I think the best option here is to make the cache opt-in and do that everywhere except project model

What do you guys think of this?